airflow/models/xcom.py

```diff
@@ -87,7 +88,7 @@ class BaseXCom(Base, LoggingMixin):
         # but it goes over MySQL's index length limit. So we instead create indexes
         # separately, and enforce uniqueness with DagRun.id instead.
         Index("idx_xcom_key", key),
-        Index("idx_xcom_ti_id", dag_id, task_id, run_id),
+        Index("idx_xcom_ti_id", dag_id, task_id, run_id, map_index),
```

Comment:
The column order here doesn't match the migrations -- this matters from one of our DBs (sqlite? mssql?)

Reply:
Shouldn't we have an FK to TaskInstance?

Reply:
There wasn’t a foreign key before. I think previously (when XCom was based on execution date) it was possible to push an XCom to a future date, so a foreign key did not make sense back then; now it’s based on run ID perhaps it makes sense to have a ti-xcom relation, but that should be a separate discussion regardless.